sender.go

```diff
@@ -148,8 +148,10 @@ func (s *Sender) Send(ctx context.Context, msg *Message, opts ...SendOption) err
 	defer span.End()
 
 	if msg.SessionID == nil {
+		s.clientMu.Lock()
```

Comment:
This should be the read lock, i.e. `s.clientMu.RLock()` and use `RUnlock()` below.

Reply:
Oh, my mistake.